src/engine/types/actions.rs

```diff
@@ -47,6 +47,33 @@ impl<T> EngineAction for CreateAction<T> {
     }
 }
 
+#[derive(Debug, PartialEq, Eq)]
+pub enum MappingCreateAction<T> {
+    Created(T),
+    Identity,
+    ValueChanged,
```

Comment:
Maybe ```ValueChanged``` should also be parameterized on ```T```. I know it makes no difference in this commit because ```T``` is bound to ```()``` in the only place it is used, for security reasons, I presume, but it makes more sense to me.

Reply:
I had also considered that so if you're also thinking that might be a good idea, I'll add it.